### PR TITLE
Use modules

### LINKED ISSUE #295: Switch to go modules
I think go modules and the tooling around it is mature enough to use now. I just switched most of my projects over, and it is significantly better. `go mod vendor` can be used to have a vendor dir for backwards compatibility and reproducibility, `gopls` works well in vscode, the `replace` directive is quite useful, and things seem quite a bit more polished than last time I tried it,

Doing this would simplify builds, make it easier to develop j-i, and simplify things in general.

cc @lvillani

### PR DESCRIPTION
**Pull request by lvillani**
This also enables the build cache on AppVeyor.

*NOTE*: explicitly using go 1.11 in go.mod so that we don't have to explicitly install 1.12 on AppVeyor.

Closes #295

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module github.com/just-install/just-install
+
+go 1.11
+
+require (
+	github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1
```
Comment (geek1011) [inline go.mod]:
```suggestion
	github.com/fatih/color v1.7.0 // indirect
	github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1
	github.com/mattn/go-colorable v0.1.1 // indirect
```

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,12 @@
+github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1 h1:PJPDf8OUfOK1bb/NeTKd4f1QXZItOX389VN3B6qC8ro=
```
Comment (geek1011) [inline go.sum]:
```suggestion
github.com/fatih/color v1.7.0 h1:DkWD4oS2D8LGGgTQ6IvwJJXSL5Vp2ffcQg58nFV38Ys=
github.com/fatih/color v1.7.0/go.mod h1:Zm6kSWBoL9eyXnKyktHP6abPY2pDugNf5KwzbycvMj4=
github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1 h1:PJPDf8OUfOK1bb/NeTKd4f1QXZItOX389VN3B6qC8ro=
```

File: go.sum [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1/go.mod h1:1NbS8ALrpOvjt0rHPNLyCIeMtbizbir8U//inJ+zuB8=
```
Comment (geek1011) [inline go.sum]:
```suggestion
github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1/go.mod h1:1NbS8ALrpOvjt0rHPNLyCIeMtbizbir8U//inJ+zuB8=
github.com/mattn/go-colorable v0.1.1 h1:G1f5SKeVxmagw/IyvzvtZE4Gybcc4Tr1tf7I8z0XgOg=
github.com/mattn/go-colorable v0.1.1/go.mod h1:FuOcm+DKB9mbwrcAfNl7/TZVBZ6rcnceauSikq3lYCQ=
github.com/mattn/go-isatty v0.0.5 h1:tHXDdz1cpzGaovsTB+TVB8q90WEokoVmfMqoVcrLUgw=
github.com/mattn/go-isatty v0.0.5/go.mod h1:Iq45c/XA43vh69/j3iqttzPXn0bhXyGjM0Hdxcsrc5s=
```

File: go.sum [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+github.com/mattn/go-runewidth v0.0.3 h1:a+kO+98RDGEfo6asOGMmpodZq4FNtnGP54yps8BzLR4=
+github.com/mattn/go-runewidth v0.0.3/go.mod h1:LwmH8dsx7+W8Uxz3IHJYH5QSwggIsqBzpuz5H//U1FU=
+github.com/ungerik/go-dry v0.0.0-20180411133923-654ae31114c8 h1:p6JzR5AMj5LyCEovRh5MOxmyuuwOEtfcVDwKqsBn41I=
+github.com/ungerik/go-dry v0.0.0-20180411133923-654ae31114c8/go.mod h1:+LeLocciSarKa1pxOY7gmBQ7dSk5nB1w1f3nvvLw0j0=
+github.com/urfave/cli v0.0.0-20180821064027-934abfb2f102 h1:Er7kUEUX12vAWCp23Uv6Nrza7kEzEm/Z77amjMT7/Lo=
+github.com/urfave/cli v0.0.0-20180821064027-934abfb2f102/go.mod h1:70zkFmudgCuE/ngEzBv17Jvp/497gISqfk5gWijbERA=
+golang.org/x/sys v0.0.0-20180921163948-d47a0f339242 h1:5DYsa+ZAwcJHjuY0Qet390sUr7qwkpnRsUNjddyc0b8=
```
Comment (geek1011) [inline go.sum]:
```suggestion
golang.org/x/sys v0.0.0-20190222072716-a9d3bda3a223 h1:DH4skfRX4EBpamg7iV4ZlCpblAHI6s6TDM39bFZumv8=
```

File: go.sum [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+golang.org/x/sys v0.0.0-20180921163948-d47a0f339242/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
```
Comment (geek1011) [inline go.sum]:
```suggestion
golang.org/x/sys v0.0.0-20190222072716-a9d3bda3a223/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
```

**Comment by lvillani** [comment]
Ran `go mod tidy` locally and force-pushed it. The web interface wanted me to accept each hunk in isolation (ugh!).

This basically re-applies the same changes I made the first time I tried to enable modules (so, no more `vendor/`) and also caches Go's build cache for slightly faster builds.

**Pull request merged**